lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1666,6 +1913,40 @@ def test_student_join_instructor_managed_team(self, *args):  # pylint: disable=u
             self.build_membership_data_raw(self.users['student_enrolled_not_on_team'].username, self.solar_team.team_id)
         )
 
+    @ddt.data(
+        ('student_masters', 400, 'is already a member'),
+        ('student_masters_not_on_team', 200, None),
+        ('student_unenrolled', 404, None),
+        ('student_enrolled', 403, None),
+        ('student_enrolled_both_courses_other_team', 403, None),
+        ('staff', 200, None),
+    )
+    @ddt.unpack
+    def test_join_organization_protected_team(self, user, expected_status, expected_message):
+        response = self.post_create_membership(
+            expected_status,
+            self.build_membership_data_raw(self.users[user].username, self.masters_only_team.team_id),
+            user=user
+        )
+        if expected_message:
+            self.assertIn(expected_message, json.loads(response.content.decode('utf-8'))['developer_message'])
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_on_team_1_private_set_1', 403),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 403),
+        ('student_on_team_2_private_set_1', 403),
+        ('student_masters', 403),
+        ('staff', 200)
+    )
+    def test_student_join_private_managed_team(self, user, expected_status):
```

Comment:
only staff can add someone to a managed team, so that looks correct.

In a test above we see that if you try to joina. nonexistant team, you 404, even if you aren't staff, so like in several other places, we should either make the 403 take precedence, so we check that before checking if the team exists, or specifically 404 to protect info from leaking

- [ ] modify bad_team to test for a staff request as well as a nonstaff request